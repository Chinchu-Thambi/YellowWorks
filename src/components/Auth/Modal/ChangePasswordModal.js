import React from 'react';
import styled from 'styled-components';

import Button from '../../Button';

import AuthContext from '../AuthContext';
import Input from '../../formElements/Input';

const Container = styled.form`
  text-align: center;
  width: 280px;
  word-wrap: normal;
`;

const ChangePasswordModal = () => {
  const authState = React.useContext(AuthContext);

  const [password, setPassword] = React.useState({
    previous: '', proposed: '', confirmation: '',
  });
  const [errorMessage, setErrorMessage] = React.useState('');

  const submitHandler = async (e) => {
    e.preventDefault();

    if (!password.previous) {
      setErrorMessage('You need to provide your current password.');
      return;
    }

    if (!password.proposed) {
      setErrorMessage('You need to set a new password.');
      return;
    }

    if (password.proposed !== password.confirmation) {
      setErrorMessage('Confirmation doesn\'t match password.');
      return;
    }

    try {
      await authState.changePassword({
        previousPassword: password.previous,
        proposedPassword: password.proposed,
      });

      authState.showHideModal({
        show: true,
        modalType: 'changeConfirmation',
        params: {
          message: 'Password successfully changed',
        },
      });
    } catch (error) {
      setErrorMessage(error.message);
    }
  };

  // remove error message on user input
  React.useEffect(() => {
    setErrorMessage('');
  }, [password]);

  return (
    <Container onSubmit={submitHandler}>
      <Input
        className="form-control"
        type="password"
        value={password.previous}
        onChange={(e) => setPassword({
          ...password,
          previous: e.target.value,
        })}
        placeholder="Current password"
      />
      <Input
        className="form-control"
        type="password"
        value={password.proposed}
        onChange={(e) => setPassword({
          ...password,
          proposed: e.target.value,
        })}
        placeholder="Enter new password"
      />
      <Input
        className="form-control"
        type="password"
        value={password.confirmation}
        onChange={(e) => setPassword({
          ...password,
          confirmation: e.target.value,
        })}
        placeholder="Re-enter new password"
      />
      <p>
        {errorMessage}
      </p>
      <Button>Confirm</Button>
    </Container>
  );
};

export default ChangePasswordModal;
