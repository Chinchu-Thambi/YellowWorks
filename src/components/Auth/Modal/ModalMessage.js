import styled from 'styled-components';

export const ErrorMessage = styled.div`
  color: red;
  text-align: center;
  font-size: 13px;
  line-height: 1.5em;
  margin-bottom: 24px;
`;

export const ErrorMessageBold = styled(ErrorMessage)`
  font-weight: ${({ theme }) => theme.fontWeight[1]};
  font-size: 16px;
`;

export const InfoText = styled.p`
  text-align: center;
  font-size: 12px;
`;

export const InfoTextBig = styled(InfoText)`
  font-size: 16px;
  font-weight: ${({ theme }) => theme.fontWeight[1]};
  margin-top: 40px;
`;

export const InfoTextVeryBig = styled(InfoText)`
  font-size: ${({ theme }) => theme.fontSizes[4]};
  color: ${({ theme }) => theme.palette.contrast[2]};
`;
export default ErrorMessage;
