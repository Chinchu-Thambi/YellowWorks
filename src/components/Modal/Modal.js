import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

import {
  ReachModal, ReachContainer, ReachContent, ReachTitle, ReachCloseButton,
} from './ModalStyled';

const Modal = ({
  children, title, isVisible, onDismiss, dismissable, ...rest
}) => {
  const toggleVisibility = () => {
    if (typeof onDismiss === 'function') {
      onDismiss(!isVisible);
    }
  };
  return (
    <ReachModal
      isOpen={isVisible}
      onDismiss={dismissable ? toggleVisibility : () => {}}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...rest}
    >
      <ReachContainer>
        {title && <ReachTitle>{title}</ReachTitle>}
        <ReachContent>
          {children}
        </ReachContent>
      </ReachContainer>

      {onDismiss !== null && (
        <ReachCloseButton
          type="button"
          onClick={toggleVisibility}
          alt="close"
        >
          <div className="text-contrast-200 hover:text-contrast-400 w-3 h-3 rounded-full flex justify-center items-center">
            <FontAwesomeIcon icon={faTimes} size="lg" />
          </div>
        </ReachCloseButton>
      )}

    </ReachModal>
  );
};

Modal.defaultProps = {
  title: undefined,
  children: [],
  dismissable: true,
  onDismiss: () => {},
  isVisible: false,
};

Modal.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  title: PropTypes.string,
  isVisible: PropTypes.bool,
  dismissable: PropTypes.bool,
  onDismiss: PropTypes.func,
};

export default Modal;
