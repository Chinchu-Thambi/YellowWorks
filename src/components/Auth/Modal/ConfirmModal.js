import React, {
  useState, useContext,
} from 'react';
import styled from 'styled-components';
import { Flex, Box } from 'rebass/styled-components';
import { NotificationManager } from 'react-notifications';

import AuthContext from '../AuthContext';
import Button from '../../Button';
import Spinner from '../../Spinner';
import Input from '../../formElements/Input';

import { InfoText, InfoTextBig, ErrorMessage } from './ModalMessage';
import {
  ModalFormContent, ModalHeader, logoSrc,
  HeaderContent, ModalBody, BodyContent,
} from './Styled';

const Container = styled.div`
  text-align: center;
`;

const DigitInput = styled(Input)`
  font-size: ${({ theme }) => theme.fontSizes[4]};
  text-align: center;
  flex-grow: 0;
  margin: auto;
`;

/* ==================================================================
* Functional Component Confirm Registration
* Confirm Registration form that contains Username and Code field, and calls
* AWS Amplify to confirm registration with emailed code to cognito.
================================================================== */
const ConfirmModal = () => {
  // Global state
  const {
    confirmSignUp, resendCode, showHideModal, login, state: { modal: { params } },
  } = useContext(AuthContext) || {};
  const { password } = params ?? {};

  // Local state
  const [username, setUsername] = useState(params?.username || '');
  const [code, setCode] = useState('');
  const [errorMessage, setErrorMessage] = useState('');
  const [isSubmitting, setIsSubmitting] = useState(false);

  // Handlers
  const handleUsername = (event) => setUsername(event.target.value.trim());
  const handleCode = (event) => setCode(event.target.value.trim().slice(0, 6));

  // Handled form submission
  const onSubmit = async (e) => {
    e.preventDefault();

    setIsSubmitting(true);
    const result = await confirmSignUp(username, code);
    if (result.success) {
      await login(username, password);
      showHideModal({
        show: true,
        loggingOut: false,
        modalType: 'success',
        params: {
          ...params,
          message: 'Account successfully activated',
        },
      });
    } else {
      setErrorMessage('Invalid authorization code');
    }
    setIsSubmitting(false);
  };

  const handleResendCode = async () => {
    setErrorMessage();
    setIsSubmitting(true);

    const { success, error } = await resendCode(username);

    if (success) {
      NotificationManager.success('Activation code was successfully submitted.');
    } else if (error?.message) {
      if (error?.code === 'UserNotFoundException') {
        setErrorMessage('Email address not found.');
      } else {
        setErrorMessage(error?.message);
      }
    }
    setIsSubmitting(false);
  };

  return (
    <ModalFormContent>
      <ModalHeader>
        <HeaderContent>
          <img src={logoSrc} width="168" alt="Yellow Logo" />
          <h2>Thanks for registering</h2>
        </HeaderContent>
      </ModalHeader>
      <ModalBody>
        <BodyContent>
          <h4>
            Thanks for creating an online account with Yellow.
            A confirmation email containing the 6 digit activation code has been sent.
          </h4>
        </BodyContent>
        <Container>
          <form onSubmit={onSubmit}>
            <Box mb={4} as="fieldset">
              <Input
                label="username"
                placeholder="Business Email"
                type="text"
                style={{ width: '100%' }}
                value={username}
                onChange={handleUsername}
              />
              <InfoText>Head over to your inbox to get the activation code.</InfoText>
              <InfoTextBig>Enter the 6 digit activation code.</InfoTextBig>
              <DigitInput
                label="code"
                placeholder="_ _ _ _ _ _"
                type="text"
                value={code}
                onChange={handleCode}
                // initial value is undefined so that user can just paste values on it
                maxLength={code !== '' ? 6 : undefined}
                size={10}
              />
              {errorMessage !== ''
                ? <ErrorMessage>{errorMessage}</ErrorMessage>
                : null}
            </Box>

            {isSubmitting && (
              <Spinner size={16} />
            )}

            <Flex
              my={3}
              flexDirection={['column', 'row']}
              alignItems={['center', 'start']}
              flexWrap="wrap"
              justifyContent="space-evenly"
            >
              <Box mr={2} mb={2}>
                <Button
                  type="submit"
                  size="sm"
                  disabled={isSubmitting}
                >ACTIVATE ACCOUNT
                </Button>
              </Box>
              <Box
                mt={[2, 0]}
                mb={2}
                as={Button}
                size="sm"
                variant="secondary"
                type="button"
                onClick={handleResendCode}
              >RESEND CODE
              </Box>
              <Button
                size="sm"
                variant="tertiary"
                type="button"
                onClick={() => {
                  showHideModal({
                    show: true,
                    modalType: 'register',
                    params: { username },
                  });
                }}
              >BACK TO SIGN UP
              </Button>
            </Flex>
          </form>
        </Container>
      </ModalBody>
    </ModalFormContent>
  );
};

ConfirmModal.propTypes = {
};

export default ConfirmModal;
