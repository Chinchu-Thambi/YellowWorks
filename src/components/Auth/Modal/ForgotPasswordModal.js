import React, {
  useState, useContext,
} from 'react';
import { Flex, Box } from 'rebass/styled-components';
import styled from 'styled-components';
import { NotificationManager } from 'react-notifications';

import AuthContext from '../AuthContext';
import Button from '../../Button';
import Input from '../../formElements/Input';
import { ErrorMessage } from './ModalMessage';

import {
  ModalFormContent, ModalHeader, logoSrc,
  HeaderContent, ModalBody, BodyContent,
} from './Styled';

const Container = styled.div`
  text-align: center;
`;

/* ==================================================================
* Functional Component ForgotPassword
* Contains Username and Password/Confirm Password field, and calls
* AWS Amplify to authenticate with cognito.
================================================================== */

export const ForgotPasswordModalResetPassword = () => {
  const {
    forgotPasswordSubmit, showHideModal, state: { modal: { params } },
  } = useContext(AuthContext) || {};

  const [newPassword, setNewPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [verificationCode, setVerificationCode] = useState('');
  const [errorMessage, setErrorMessage] = useState('');

  const { username } = params || {};

  const handleVerificationCode = (event) => setVerificationCode(event.target.value);
  const handleNewPassword = (event) => setNewPassword(event.target.value);
  const handleConfirmPassword = (event) => setConfirmPassword(event.target.value);

  const handleSubmit = async (e) => {
    e.preventDefault();

    if (!verificationCode) {
      setErrorMessage('You need to fill the verification code.');
      return;
    }

    if (!newPassword) {
      setErrorMessage('You need to fill the new password.');
      return;
    }

    if (newPassword !== confirmPassword) {
      setErrorMessage('Confirmation needs to match password.');
      return;
    }

    const result = await forgotPasswordSubmit(
      username, verificationCode, newPassword,
    );

    if (result.error) {
      setErrorMessage(result.error.message);
      return;
    }

    showHideModal({
      show: true,
      modalType: 'success',
      params: {
        username,
        password: newPassword,
        message: 'Your password has been reset.',
      },
    });
  };

  return (
    <>
      <ModalFormContent>
        <ModalHeader>
          <HeaderContent>
            <img src={logoSrc} width="168" alt="Yellow Logo" />
            <h2>Reset Password</h2>
          </HeaderContent>
        </ModalHeader>
        <ModalBody>
          <BodyContent>
            <h4>
              We have sent you an email with a 6 digit verification code.
              Please enter this code, followed by your new password.
            </h4>
          </BodyContent>
          <Container>
            <form>
              <p>{username}</p>
              <Box as="fieldset" width={0.5} mx="auto" mb={3}>
                <Input
                  label="verificationCode"
                  placeholder="Verification Code"
                  type="text"
                  style={{ width: '100%' }}
                  value={verificationCode}
                  onChange={handleVerificationCode}
                />
                <Box my={2}>
                  <Input
                    label="New Password"
                    placeholder="New Password"
                    type="password"
                    style={{ width: '100%' }}
                    value={newPassword}
                    onChange={handleNewPassword}
                  />
                </Box>
                <Input
                  label="Confirm"
                  placeholder="Confirm Password"
                  type="password"
                  style={{ width: '100%' }}
                  value={confirmPassword}
                  onChange={handleConfirmPassword}
                />
              </Box>

              {errorMessage && (
                <p>{errorMessage}</p>
              )}

              <div>
                <Box mr={2} mb={2}>
                  <Button
                    size="sm"
                    onClick={handleSubmit}
                  >CONFIRM NEW PASSWORD
                  </Button>
                </Box>
                <Button
                  size="sm"
                  type="button"
                  variant="secondary"
                  onClick={() => {
                    showHideModal({
                      show: true,
                      modalType: 'fpemail',
                      params: {
                        ...params,
                        username,
                      },
                    });
                  }}
                >BACK
                </Button>
              </div>
            </form>
          </Container>
        </ModalBody>
      </ModalFormContent>
    </>
  );
};

ForgotPasswordModalResetPassword.propTypes = {
};

export const ForgotPasswordModalGetEmail = () => {
  const {
    forgotPassword, showHideModal, state: { modal: { params } },
  } = useContext(AuthContext) || {};
  const [username, setUsername] = useState(params?.username);
  const [errorMessage, setErrorMessage] = useState('');

  const handleUsername = (event) => setUsername(event.target.value.trim());

  const handleSubmit = async (e) => {
    e.preventDefault();

    if (!username) {
      setErrorMessage('You need to fill in the email address.');
      return;
    }

    const response = await forgotPassword(username);

    if (response.error) {
      if (response.error.code === 'UserNotFoundException') {
        setErrorMessage('Email address not found.');
      } else {
        setErrorMessage(response.error.message);
      }
    } else {
      NotificationManager.success('Reset password code was successfully submitted.');
      showHideModal({
        show: true,
        modalType: 'fpreset',
        params: { username },
      });
    }
  };

  return (
    <ModalFormContent>
      <ModalHeader>
        <HeaderContent>
          <img src={logoSrc} width="168" alt="Yellow Logo" />
          <h2>Reset password</h2>
        </HeaderContent>
      </ModalHeader>
      <ModalBody>
        <BodyContent>
          <h4>
            Enter the email address associated with your account, and we will email you a code
            to reset your password
          </h4>
        </BodyContent>
        <Container>
          <form onSubmit={handleSubmit}>
            <Box mb={4}>
              <Input
                label="Business Email"
                placeholder="Business Email"
                type="text"
                style={{ width: '100%' }}
                value={username}
                onChange={handleUsername}
              />
            </Box>
            {errorMessage !== ''
              ? (
                <>
                  <ErrorMessage>
                    {errorMessage}
                  </ErrorMessage>
                  <br />
                </>
              )
              : null}

            <Flex
              my={3}
              flexDirection={['column', 'row']}
              alignItems={['center', 'start']}
              flexWrap="wrap"
              justifyContent="space-evenly"
            >
              <Button
                size="sm"
              >SEND CODE
              </Button>
              <Box
                mt={[2, 0]}
                mb={2}
                as={Button}
                size="sm"
                variant="secondary"
                type="button"
                onClick={() => {
                  showHideModal({
                    show: true,
                    modalType: 'fpreset',
                    params: {
                      ...params,
                      username,
                    },
                  });
                }}
              >ALREADY GOT A CODE
              </Box>
              <Button
                size="sm"
                variant="tertiary"
                type="button"
                onClick={() => {
                  showHideModal({
                    show: true,
                    modalType: 'login',
                    params: { username },
                  });
                }}
              >BACK TO SIGN IN
              </Button>
            </Flex>
          </form>
        </Container>
      </ModalBody>
    </ModalFormContent>
  );
};

ForgotPasswordModalGetEmail.propTypes = {
};

export const SystemForgotPasswordModal = () => {
  const {
    forgotPassword, showHideModal, state: { modal: { params } },
  } = useContext(AuthContext) || {};

  const [username, setUsername] = useState(params?.username);

  return (
    <>
      <ModalFormContent>
        <ModalHeader>
          <HeaderContent>
            <img src={logoSrc} width="168" alt="Yellow Logo" />
            <h2>Hello!</h2>
          </HeaderContent>
        </ModalHeader>
        <ModalBody>
          <BodyContent>
            <p>
              It’s time for you to update your password with Yellow.
              It’s pretty easy, we’ll email you a 6-digit code to verify your account.
              All you need to do is choose a new password, input the code and you’re all set!
            </p>
          </BodyContent>
          <Container>
            <form>
              <Input
                label="Business Email"
                placeholder="Business Email"
                type="text"
                style={{ width: '100%' }}
                value={username}
                onChange={(e) => setUsername(e.target.value.trim())}
              />
              <Flex justifyContent="space-evenly" mt={3}>
                <Button
                  size="sm"
                  variant="tertiary"
                  outline="true"
                  onClick={async (e) => {
                    e.preventDefault();
                    showHideModal({
                      show: true,
                      modalType: 'login',
                    });
                  }}
                >BACK TO SIGN IN
                </Button>
                <Button
                  size="sm"
                  onClick={async (e) => {
                    e.preventDefault();
                    if (await forgotPassword(username)) {
                      showHideModal({
                        show: true,
                        modalType: 'fpreset',
                        params: { username },
                      });
                    }
                  }}
                >SEND CODE
                </Button>
              </Flex>
            </form>
          </Container>
        </ModalBody>
      </ModalFormContent>
    </>
  );
};
