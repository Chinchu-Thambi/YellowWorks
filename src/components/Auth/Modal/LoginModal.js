/* eslint-disable max-len */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, {
  useState, useContext,
} from 'react';
import { Flex, Box } from 'rebass/styled-components';
import styled from 'styled-components';
import { navigate } from 'gatsby';

import AuthContext from '../AuthContext';
import Button from '../../Button';
import Input from '../../formElements/Input';

import SocialLoginButtons from './components/SocialLoginButtons';
import { ErrorMessage, ErrorMessageBold } from './ModalMessage';
import {
  ModalFormContent, ModalHeader, logoSrc,
  HeaderContent, ModalBody, Separator,
} from './Styled';

const Fieldset = styled.fieldset`
  && {
    clear: both;
    margin-top: ${({ theme }) => theme.space[3]};
  }
`;

const StyledButton = styled(Button)`
  display: block;
  margin: 0 auto ${({ theme }) => theme.space[4]};
  width: 150px;
  text-align: center;
`;

const LinkButton = styled.button`
  cursor: pointer;
  border: none;
  padding: 0;
  margin-bottom: 24px;
  background: none;
`;

const Agreement = styled.div`
  text-align: center;
  font-size: ${({ theme }) => theme.fontSizes[1]};
  line-height: 1.5;
`;

/* ==================================================================
* Functional Component Login
* Login form that contains Username and Password field, and calls
* AWS Amplify to authenticate with cognito.
================================================================== */
const LoginModal = () => {
  // Global state
  const {
    login, loginYellow, state: { modal: { params, target } },
    showHideModal,
  } = useContext(AuthContext) || {};


  // Local state
  const [username, setUsername] = useState(params?.username);
  const [password, setPassword] = useState('');
  const [passwordVisibility, setPasswordVisibility] = useState(true);
  const [errorMessage, setErrorMessage] = useState('');

  // Handlers
  const handleUsername = (event) => {
    event.preventDefault();
    const { value } = event.target;

    setUsername(value.trim());
    const yellowDomain = value?.endsWith('@yellow.co.nz');
    if (yellowDomain) {
      setPasswordVisibility(false);
    } else {
      setPasswordVisibility(true);
    }
  };

  const handlePassword = (event) => setPassword(event.target.value);

  const onSubmit = async (e) => {
    e.preventDefault();

    // Check if it is a yellow domain. If so, do a federated signin with Yellow
    if (username.endsWith('@yellow.co.nz')) {
      loginYellow();
    } else {
      const { success, error } = await login(username, password);

      if (success) {
        if (target) navigate(target);

        // case for ristretto users migrated to nw cognito:
      } else if (error && error.code && error.code.includes('PasswordResetRequiredException')) {
        showHideModal({
          show: true,
          modalType: 'systemForgotPassword',
          params: { username },
        });
      } else {
        setErrorMessage('Please check your email and password. If your previously signed up with Google or Facebook, click the links to access the Yellow account');
      }
    }
  };

  const handleRegister = async (e) => {
    e.preventDefault();
    showHideModal({
      show: true,
      modalType: 'register',
    });
  };

  const handleForgotPassword = async (e) => {
    e.preventDefault();
    showHideModal({
      show: true,
      modalType: 'fpemail',
      params: { username },
    });
  };

  return (
    <ModalFormContent>
      <ModalHeader>
        <HeaderContent>
          <img src={logoSrc} width="168" alt="Yellow Logo" />
          <h2>Sign in</h2>
        </HeaderContent>
      </ModalHeader>
      <ModalBody>
        {errorMessage !== '' && (
          <ErrorMessage>
            <ErrorMessageBold>
              Unable to login?
            </ErrorMessageBold>
            <p>{errorMessage}</p>
          </ErrorMessage>
        )}

        <form onSubmit={onSubmit}>
          <SocialLoginButtons />
          <Separator>or</Separator>
          <Fieldset>
            <Input
              label="username"
              placeholder="Business Email"
              type="email"
              style={{ width: '100%' }}
              value={username}
              onChange={handleUsername}
              required
            />
            <Box my={2}>
              {passwordVisibility && (
                <Input
                  label="password"
                  placeholder="Password"
                  type="password"
                  style={{ width: '100%' }}
                  value={password}
                  onChange={handlePassword}
                  required
                />
              )}
            </Box>
          </Fieldset>

          <StyledButton size="sm">SIGN IN</StyledButton>
        </form>
        <Flex justifyContent="space-around" width={1} mb={3}>
          <LinkButton onClick={handleRegister} tabIndex={0}>Create Account</LinkButton>
          <LinkButton style={{ float: 'right' }} onClick={handleForgotPassword} tabIndex={0}>Forgot Password</LinkButton>
        </Flex>
        <Agreement>
          By signing up I agree to Yellow Terms of Use, Privacy Policy and (where applicable) the Terms of Business
        </Agreement>

      </ModalBody>
    </ModalFormContent>
  );
};

LoginModal.propTypes = {
};

export default LoginModal;
