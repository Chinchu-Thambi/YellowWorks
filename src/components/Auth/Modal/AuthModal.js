import React, { useContext } from 'react';
import { navigate } from 'gatsby';

import AuthContext from '../AuthContext';
import Modal from '../../Modal';

import LoginModal from './LoginModal';
import RegisterModal from './RegisterModal';
import ConfirmModal from './ConfirmModal';
import {
  ForgotPasswordModalGetEmail,
  ForgotPasswordModalResetPassword,
  SystemForgotPasswordModal,
} from './ForgotPasswordModal';
import SuccessModal from './SuccessModal';
import ChangePasswordModal from './ChangePasswordModal';
import RegistrationsDisabled from './RegistrationsDisabled';
import ExistingProfile from './ExistingProfile';

const AuthModal = () => {
  const {
    state: {
      modal: {
        showAuthModal, modalType, params, dismissTarget, dismissable,
      } = {},
    } = {},
    showHideModal,
  } = useContext(AuthContext) || {};

  const handleClose = async () => {
    if (dismissTarget) navigate(dismissTarget);

    showHideModal({ show: false });
  };

  let titleText;
  let modalContent;

  switch (modalType) {
    case 'login':
      modalContent = <LoginModal />;
      break;
    case 'register':
      modalContent = <RegisterModal />;
      break;
    case 'confirm':
      modalContent = <ConfirmModal />;
      break;
    case 'fpemail':
      modalContent = <ForgotPasswordModalGetEmail />;
      break;
    case 'fpreset':
      modalContent = <ForgotPasswordModalResetPassword />;
      break;
    case 'success':
      modalContent = <SuccessModal message={params?.message} />;
      break;
    case 'changePassword':
      titleText = 'Change Password';
      modalContent = <ChangePasswordModal />;
      break;
    case 'changeConfirmation':
      titleText = 'Thank you';
      modalContent = <SuccessModal message={params?.message} />;
      break;
    case 'systemForgotPassword':
      modalContent = <SystemForgotPasswordModal />;
      break;
    case 'registrationsDisabled':
      modalContent = <RegistrationsDisabled />;
      break;
    case 'existingProfile':
      titleText = 'You have existing Yellow Profiles on this account';
      modalContent = <ExistingProfile profiles={params?.profiles} />;
      break;
    default:
      titleText = null;
      modalContent = null;
      break;
  }

  return (
    <Modal
      isVisible={Boolean(showAuthModal)}
      onDismiss={handleClose}
      dismissable={dismissable}
      title={titleText}
    >
      {modalContent}
    </Modal>
  );
};

export default AuthModal;
