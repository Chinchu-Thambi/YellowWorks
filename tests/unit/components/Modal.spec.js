/* globals describe, it, expect */
import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import Modal from '../../../src/components/Modal';

const theme = {
  fonts: {
    mono: 'fontMono',
  },
  zIndex: {
    sticky: 1000,
  },
  breakpoints: [
    1,
  ],
  colors: {
    white: '',
  },
  palette: {
    base: [0],
    contrast: [0, 1, 2, 3, 4, 5, 6],
    brand: [0, 1, 2, 3, 4, 5, 6],
  },
  buttons: {
    primary: {
      borderRadius: '',
    },
  },
  fontSizes: [0, 1, 2, 3, 4, 5],
  space: [0, 1, 2, 3, 4, 5],
  shadows: {
    small: 0,
    large: 1,
  },
};

describe('Modal', () => {
  const modalChildren = (
    <div data-testid="child">I am a children</div>
  );

  it('renders without crashing', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Modal
          isVisible
          onDismiss={() => { }}
        >
          {modalChildren}
        </Modal>
      </ThemeProvider>,
    );

    expect(wrapper).toBeDefined();
    expect(wrapper).toBeTruthy();

    wrapper.unmount();
  });

  it('has specific containers', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Modal
          title="testing title"
          isVisible
          onDismiss={() => { }}
        >
          {modalChildren}
        </Modal>
      </ThemeProvider>,
    );

    expect(wrapper.find('h2').text()).toBe('testing title');
    expect(wrapper.find('[data-testid="child"]').text()).toBe('I am a children');

    wrapper.unmount();
  });

  it('should have a close button', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Modal
          isVisible
          onDismiss={() => { }}
        >
          {modalChildren}
        </Modal>
      </ThemeProvider>,
    );

    expect(wrapper.find('button')).toBeTruthy();
    expect(wrapper.find('button').prop('alt')).toBe('close');

    wrapper.unmount();
  });

  it('should render without a title', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Modal
          isVisible
          onDismiss={() => { }}
        >
          {modalChildren}
        </Modal>
      </ThemeProvider>,
    );

    expect(wrapper.find('h2')).toHaveLength(0);

    wrapper.unmount();
  });

  it('should match snapshot', () => {
    const wrapper = mount(<ThemeProvider theme={theme}><Modal /></ThemeProvider>);
    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });
});
