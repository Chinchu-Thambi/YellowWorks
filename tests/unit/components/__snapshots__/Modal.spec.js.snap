// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
<Component
  theme={
    Object {
      "breakpoints": Array [
        1,
      ],
      "buttons": Object {
        "primary": Object {
          "borderRadius": "",
        },
      },
      "colors": Object {
        "white": "",
      },
      "fontSizes": Array [
        0,
        1,
        2,
        3,
        4,
        5,
      ],
      "fonts": Object {
        "mono": "fontMono",
      },
      "palette": Object {
        "base": Array [
          0,
        ],
        "brand": Array [
          0,
          1,
          2,
          3,
          4,
          5,
          6,
        ],
        "contrast": Array [
          0,
          1,
          2,
          3,
          4,
          5,
          6,
        ],
      },
      "shadows": Object {
        "large": 1,
        "small": 0,
      },
      "space": Array [
        0,
        1,
        2,
        3,
        4,
        5,
      ],
      "zIndex": Object {
        "sticky": 1000,
      },
    }
  }
>
  <Modal
    dismissable={true}
    isVisible={false}
    onDismiss={[Function]}
  >
    <Styled(Dialog)
      isOpen={false}
      onDismiss={[Function]}
    >
      <Dialog
        className="sc-bdnylx QZZVF"
        isOpen={false}
        onDismiss={[Function]}
      >
        <ForwardRef
          isOpen={false}
          onDismiss={[Function]}
        >
          <Component
            didMount={[Function]}
            getInitialState={[Function]}
            getRefs={[Function]}
          />
        </ForwardRef>
      </Dialog>
    </Styled(Dialog)>
  </Modal>
</Component>
`;
